### PR TITLE
Search: implement stale queue alerts

### PR DESCRIPTION
**Pull request by pschoffer**
## Description

Sends an alert to cantina channel if the indexing queue gets stale.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- [x] This change has relevant documentation additions / updates (if applicable).

## Steps to Test

Couldn't figure that out:(

### CODE REVIEW

File: search/includes/classes/class-search.php [hunk 1]
```diff
@@ -614,24 +628,24 @@ public static function record_ratelimited_query_stat() {
...
-
-		$url = $this->get_current_host();
-		$per_site_stat = $this->get_statsd_prefix( $url, $statsd_mode, FILES_CLIENT_SITE_ID, $statsd_index_name );
-
-		$statsd = new \Automattic\VIP\StatsD();
-
-		$statsd->gauge( $per_site_stat, $average_wait_time );
+		if ( $average_wait_time > self::STALE_QUEUE_WAIT_LIMIT ) {
+			$message = sprintf(
+				'Average index queue time for application %d, exceeded %d seconds with %d seconds',
```
Comment (nickdaugherty) [inline search/includes/classes/class-search.php:642]:
```suggestion
				'Average index queue wait time for application %d is currently %d seconds',
```

I would suggest putting the current queue wait time before the threshold, for ease of reading. We can also just leave the threshold off entirely here. 

We should add the site's domain though, like we do on other alerts.

File: search/includes/classes/class-search.php [hunk 2]
```diff
@@ -15,12 +15,16 @@ class Search {
 	private const QUERY_COUNT_TTL = 300; // 5 minutes in seconds.
 	private const MAX_SEARCH_LENGTH = 255;
 	private const DISABLE_POST_META_ALLOW_LIST = array();
+	private const STALE_QUEUE_WAIT_LIMIT = 3600; // 1 hour in seconds
+	private const STALE_QUEUE_ALERT_SLACK_CHAT = '#vip-platform-cantina';
```
Comment (nickdaugherty) [inline search/includes/classes/class-search.php:19]: Do we want this channel, or should we use our other existing ES alerts channel (cc @brandon-m-skinner for thoughts).

**Review by nickdaugherty — APPROVED** [review]
Looks good, just left two small suggestions (well one is a question really).

File: search/includes/classes/class-search.php (hunk 1 above)
Comment (pschoffer) [reply search/includes/classes/class-search.php]: Fixed

File: search/includes/classes/class-search.php (hunk 2 above)
Comment (pschoffer) [reply search/includes/classes/class-search.php:19]: I managed to test somehow with Nick's help. Still, the channel question is outstanding.

Comment (rinatkhaziev) [reply search/includes/classes/class-search.php:19]: Strongly vote for the other channel.

**Review by rinatkhaziev — APPROVED** [review]
Thanks, looking good!

**Comment by pschoffer** [comment]
stacks-r1231

**Pull request merged**